servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/Completable.java

```diff
@@ -1387,7 +1388,23 @@ public final Completable liftAsync(CompletableOperator operator) {
      * {@link Subscriber}.
      */
     public final Completable publishOn(Executor executor) {
-        return PublishAndSubscribeOnCompletables.publishOn(this, executor);
+        return PublishAndSubscribeOnCompletables.publishOn(this, Boolean.TRUE::booleanValue, executor);
+    }
+
+    /**
+     * Creates a new {@link Completable} that will use the passed {@link Executor} to invoke all {@link Subscriber}
+     * methods.
+     * This method does <strong>not</strong> override preceding {@link Executor}s, if any, specified for {@code this}
+     * {@link Completable}. Only subsequent operations, if any, added in this execution chain will use this
+     * {@link Executor}.
+     *
+     * @param offload Offload only if supplier returns true otherwise continue execution on the same thread.
```

Comment:
also clarify the return value serves as a hint. for example even if this returns `{@code false}` offloading may still be done (e.g. to preserve ordering).